integration/run/run_test.go

```diff
@@ -932,6 +932,304 @@ func TestDeployParam(t *testing.T) {
+						Default: "512Mi",
+						Max:     "1Gi",
+						Min:     "512Mi",
+					},
+					SupportedRegions: []string{apiv1.LocalRegion},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "acorn-test-custom-2",
+						Namespace: c.GetNamespace(),
+					},
+					Default: true,
+					Memory: adminv1.ComputeClassMemory{
+						Default: "513Mi",
+						Max:     "1Gi",
+						Min:     "512Mi",
+					},
+					SupportedRegions: []string{"local2"},
+				},
+			},
+			expected: map[string]v1.Scheduling{"simple": {
+				Requirements: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("512Mi")},
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("512Mi"),
+					},
+				},
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      tolerations.WorkloadTolerationKey,
+						Operator: corev1.TolerationOpExists,
+					},
+				}},
+			},
+			waitFor: func(obj *apiv1.App) bool {
+				return obj.Status.Condition(v1.AppInstanceConditionParsed).Success &&
+					obj.Status.Condition(v1.AppInstanceConditionScheduling).Success
+			},
+		},
+		{
+			name:              "acornfile-computeclass-wrong-region",
+			testDataDirectory: "./testdata/computeclass",
+			region:            apiv1.LocalRegion,
+			computeClasses: []adminv1.ProjectComputeClassInstance{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "acorn-test-custom-2",
+						Namespace: c.GetNamespace(),
+					},
+					Default: true,
+					Memory: adminv1.ComputeClassMemory{
+						Default: "512Mi",
+						Max:     "1Gi",
+						Min:     "512Mi",
+					},
+					SupportedRegions: []string{apiv1.LocalRegion},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "acorn-test-custom",
+						Namespace: c.GetNamespace(),
+					},
+					Default: true,
+					Memory: adminv1.ComputeClassMemory{
+						Default: "513Mi",
+						Max:     "1Gi",
+						Min:     "512Mi",
+					},
+					SupportedRegions: []string{"local2"},
+				},
+			},
+			fail: true,
+		},
+	}
+
+	for _, tt := range checks {
+		clusterObjects := make([]crClient.Object, 0, len(tt.computeClasses))
+		for _, pcc := range tt.computeClasses {
+			cc := adminv1.ClusterComputeClassInstance(pcc)
+			clusterObjects = append(clusterObjects, cc.DeepCopy())
+		}
+		projectObjects := make([]crClient.Object, 0, len(tt.computeClasses))
+		for _, cc := range tt.computeClasses {
+			projectObjects = append(projectObjects, cc.DeepCopy())
+		}
+		// Perform the same test cases on both Project and Cluster ComputeClasses
+		for kind, computeClasses := range map[string][]crClient.Object{"projectcomputeclass": projectObjects, "clustercomputeclass": clusterObjects} {
+			testcase := fmt.Sprintf("%v-%v", kind, tt.name)
+			t.Run(testcase, func(t *testing.T) {
+				if !tt.noComputeClass {
+					for _, computeClass := range computeClasses {
+						if err := kclient.Create(ctx, computeClass); err != nil {
+							t.Fatal(err)
```

Comment:
nit: you can use `require.NoError(t, err)` to end the testcase early.

Reply:
The test structure was lifted from the function below this one so I'd prefer not to change much about it other than the looping. I thought about combining the two blocks together as well but figured it would be easier to review if it was just it's own function